LvivCompany.Bookstore/LvivCompany.Bookstore.Web/Controllers/BookDetailController.cs

```diff
@@ -19,20 +20,11 @@ public BookDetailController(IRepo<Book> bookRepo)
             _bookRepo = bookRepo;
         }
 
-        public async Task<IActionResult> Index()
+        public async Task<IActionResult> Index(int id)
         {
-            var book = await _bookRepo.GetAsync(8);
-            var config = new MapperConfiguration(cfg => {
-                cfg.CreateMap<Book, BookDetailViewModel>()
-                .ForMember(bv => bv.Author, b => b.MapFrom(
-                a => (a.BookAuthors.First().Author.FirstName + " "+  a.BookAuthors.First().Author.LastName)  ))
-                .ForMember(bv => bv.Category, b => b.MapFrom(
-                a => a.Category.Name))
-                .ForMember(bv => bv.Publisher, b => b.MapFrom(
-                a => a.Publisher.Name));
-            });
+            var book = await _bookRepo.GetAsync(id);
 
-            IMapper mapper = config.CreateMapper();
+            IMapper mapper = Mappers.MapperForBook().CreateMapper();
```

Comment:
Not good for unit testing. Better way to implement mappers like below and then inject them using dependency injection:

**Example for Book <=> BookInfo**
```CSharp
IMapper<TSource, TDestination>
{
    TDestination Map(TSource entity);
    TSource Map(TDestination entity);
}

BookMapper : IMapper<Book, BookInfo>
{
    private IMapper mapper;

    public BookMapper()
    {
        var config = new MapperConfiguration(c => { c.CreateMap<Book, BookInfo>(); });
        mapper = config.CreateMapper();
    }

    BookInfo Map(Book entity) => mapper.Map<Book, BookInfo>(entity);

    Book Map(BookInfo entity) => mapper.Map<BookInfo, Book>(entity);
}
```